File: webflux-dialogue/src/main/java/com/study/webflux/rag/infrastructure/monitoring/config/TtsEndpointMetricsRegistrar.java
package com.study.webflux.rag.infrastructure.monitoring.config;

import org.springframework.stereotype.Component;

import com.study.webflux.rag.infrastructure.dialogue.adapter.tts.loadbalancer.TtsLoadBalancer;
import jakarta.annotation.PostConstruct;

/**
 * 애플리케이션 시작 시 TTS 엔드포인트 메트릭 Gauge를 등록합니다.
 */
@Component
public class TtsEndpointMetricsRegistrar {

	private final TtsLoadBalancer ttsLoadBalancer;
	private final TtsBackpressureMetrics ttsBackpressureMetrics;

	public TtsEndpointMetricsRegistrar(TtsLoadBalancer ttsLoadBalancer,
		TtsBackpressureMetrics ttsBackpressureMetrics) {
		this.ttsLoadBalancer = ttsLoadBalancer;
		this.ttsBackpressureMetrics = ttsBackpressureMetrics;
	}

	@PostConstruct
	public void register() {
		ttsBackpressureMetrics.registerEndpoints(ttsLoadBalancer.getEndpoints());
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/infrastructure/monitoring/config/TtsEndpointMetricsRegistrar.kt
package com.miyou.app.infrastructure.monitoring.config

import com.miyou.app.infrastructure.dialogue.adapter.tts.loadbalancer.TtsLoadBalancer
import jakarta.annotation.PostConstruct
import org.springframework.stereotype.Component

@Component
class TtsEndpointMetricsRegistrar(
    private val ttsLoadBalancer: TtsLoadBalancer,
    private val ttsBackpressureMetrics: TtsBackpressureMetrics,
) {
    @PostConstruct
    fun register() {
        ttsBackpressureMetrics.registerEndpoints(ttsLoadBalancer.endpoints)
    }
}
